apps/api/views/chat.py

```diff
@@ -55,13 +62,63 @@ def check_experiment_access(experiment, participant_id):
     return None
 
 
-def check_session_access(session):
+def handle_embedded_widget_auth(request, experiment_id=None, session=None):
```

Comment:
Nitty request: May I request that this method be moved to an "auth.py" (or whatever makes sense) file instead?

Reply:
Generally agree with @SmittieC. I think this should just return the channel (or set the channel on the request). Returning the boolean is equivalent to the channel being returned / set on the request.

Extracting the remote ID should be handed directly in the view or in a separate utility.